woocommerce/woocommerce
# Blocks E2E: Align ESLint and TS configs with Gutenberg

## What?

Align our ESlint and TS configs with Gutenberg's since its E2E utils are our testing base. The updated rules have already caught quite a few instances that were causing our tests to be flaky, like missing `await`s. See the changes for more details.

This PR also fixes the `Unable to resolve path to module` errors for the `@woocommerce/e2e-playwright-utils` and `@woocommerce/e2e-utils` packages (which aren't really packages but just TS path aliases): 

![image](https://github.com/woocommerce/woocommerce/assets/1451471/7c038442-08e3-4038-9e0d-e39d20572554)

After fixing the linter errors, some tests started to fail or became flaky for other reasons, so I've hopefully fixed them. Some of the fixes, however, might be unrelated to the changes in this PR, but that's just the nature of the flaky tests - they appear randomly. 🤷 


## How to test

Ensure the aforementioned unresolved path errors do not occur locally. Ensure there are no linter errors locally (warnings are OK). All the CI lint checks should pass.
